protoc-gen-swagger/genswagger/template.go

```diff
@@ -408,31 +408,35 @@ func schemaOfField(f *descriptor.Field, reg *descriptor.Registry, refs refMap) s
 		}
 	}
 
+	ret := swaggerSchemaObject{}
+
 	switch aggregate {
 	case array:
-		return swaggerSchemaObject{
+		ret = swaggerSchemaObject{
 			schemaCore: schemaCore{
 				Type:  "array",
 				Items: (*swaggerItemsObject)(&core),
 			},
 		}
 	case object:
-		return swaggerSchemaObject{
+		ret = swaggerSchemaObject{
 			schemaCore: schemaCore{
 				Type: "object",
 			},
 			AdditionalProperties: &swaggerSchemaObject{Properties: props, schemaCore: core},
 		}
 	default:
-		ret := swaggerSchemaObject{
+		ret = swaggerSchemaObject{
 			schemaCore: core,
 			Properties: props,
 		}
-		if j, err := extractJSONSchemaFromFieldDescriptor(fd); err == nil {
-			updateSwaggerObjectFromJSONSchema(&ret, j)
-		}
-		return ret
 	}
+
+	if j, err := extractJSONSchemaFromFieldDescriptor(fd); err == nil {
+		updateSwaggerObjectFromJSONSchema(&ret, j)
+	}
+
+	return ret
```

Comment:
Is this change necessary? It changes the way we handle `object` and `array` aggregates for no obvious reason to me.

Reply:
This change allows add field annotations to objects and arrays, before these changes, all field annotations added to object or arrays were ignored.